hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -511,10 +567,22 @@ public void makeQualified(FileStatus status, URI uri, Path path,
 
     incrementCounter(Statistic.OBJECTS_LIST);
     OFSPath ofsPath = new OFSPath(pathStr);
-    // TODO: Subject to change in HDDS-2928.
+
+    // listStatus root
+    if (ofsPath.getVolumeName().isEmpty()
```

Comment:
Can we add OFSPath helpers as suggested above?

Reply:
done in ada1ac4b0b0164c71e912a70c495c269887cde09